packages/circuit-ui/components/Dialog/Dialog.tsx

```diff
@@ -258,17 +268,26 @@ export const Dialog = forwardRef<HTMLDialogElement, DialogProps>(
     }, [animationDurationRef.current, onCloseStart]);
 
     const handleOutsideClick = useCallback(() => {
-      lastFocusedElementRef.current = null;
-      handleDialogClose();
-    }, [handleDialogClose]);
+      if (!preventOutsideClickClose) {
```

Comment:
This check is redundant since `useClickOutside` is only enabled when `preventOutsideClickClose` is truthy on line 285.